hamocc/powach.F90

```diff
@@ -309,29 +317,36 @@ subroutine powach(kpie,kpje,kpke,kbnd,prho,omask,psao,lspin)
            solrat(i,k) = sedlay(i,j,k,issso12) * dissot/(1. + dissot*sediso(i,k))
            posol = sediso(i,k)*solrat(i,k)
            aerob(i,k) = posol*umfa     !this has P units: kmol P/m3 of pore water
-#ifdef cisonew
-           rato13 = sedlay(i,j,k,issso13) / (sedlay(i,j,k,issso12) + safediv)
-           rato14 = sedlay(i,j,k,issso14) / (sedlay(i,j,k,issso12) + safediv)
-           poso13 = posol*rato13
-           poso14 = posol*rato14
-           aerob13(i,k) = poso13*umfa  !this has P units: kmol P/m3 of pore water
-           aerob14(i,k) = poso14*umfa  !this has P units: kmol P/m3 of pore water
-#endif
+
            sedlay(i,j,k,issso12) = sedlay(i,j,k,issso12) - posol
            powtra(i,j,k,ipowaph) = powtra(i,j,k,ipowaph) + posol*umfa
            powtra(i,j,k,ipowno3) = powtra(i,j,k,ipowno3) + posol*rnit*umfa
            powtra(i,j,k,ipowaox) = sediso(i,k)
-#ifdef cisonew
-           sedlay(i,j,k,issso13) = sedlay(i,j,k,issso13) - poso13
-           sedlay(i,j,k,issso14) = sedlay(i,j,k,issso14) - poso14
-           ! is this correct? no correspondance in the lines above
-           powtra(i,j,k,ipowc13) = powtra(i,j,k,ipowc13) + poso13*umfa
-           powtra(i,j,k,ipowc14) = powtra(i,j,k,ipowc14) + poso14*umfa
-#endif
         endif
      enddo
   enddo
 
+  if (use_cisonew) then
```

Comment:
Dear Mariana, thanks for the efforts! First skimming through the changes, this code separation cannot work, since `posol` is updated based on `solrat` and `sediso` in the loop above, similarly `umfa`. So their calculations need to be repeated (NOT updating `solrat`, since that would violate the scheme) - or solved differently. Similar issues might happen elsewhere (I just started to have a first quick glance on the changes).

Reply:
@jmaerz - Good catch! You are totally right. This was the one file where I had tried to separate the loops thinking of vectorization - and I never went back and fixed things. But there are already ifs in there to begin with. I have pushed back a fix that I think will now work. I don't think I've changed anything else that would have the same impact. I could not test this by turning on cisonew or other ifdefs since the code would not compile with them - so its not clear to me how baselines without these new changes could be done.

That said - I don't want to add more confusion to the development process right now. So I just wanted to have the code out there to start a discussions. I have confirmed that with the changes I have - the standard HAMOCC output is bit-for-bit with the new regression testing where I output a lot more HAMOCC fields. I'm happy to go over this in a BLOM development meeting when its the right time to look into this further.

Reply:
Dear @mvertens , I suspect that some of the compilation errors stem from introducing the explicit use statements a while back. `boxatm` is indeed not present (since a long while)  - @TomasTorsvik: can we recover this from the repo? @mvertens: In order to make the code compilable again: I could introduce the correct `use` statements in a fork of your repo - asking then for a pull into your branch, if that would help (and if you agree). I will continue skimming through the code a bit later.

Reply:
@jmaerz - that sounds like a great plan. Thanks so much!
 I have added you as a collaborator to my fork https://github.com/mvertens/BLOM.git.

I think what we want to do is to have you branch off of my branch - feature/refactor_buildnml - and make that compile. 
Using this branch as a starting point would make it very easy for me to add new regression tests that actually look at the output from the different #ifdefs and I could use that to compare to my branch. This would also be a great way to add more regression tests to blom in different test categories that could be exercised at different times. 
Then you could either push that branch to my fork or simply put that branch in your fork. 
To do this - you would do the following
```
> git clone https://github.com/mvertens/BLOM
> git checkout feature/refactor_buildnml
> git checkout -b feature/fix_compilations
> do your edits and commit
> git push origin feature/refactor_buildnml
```

Does my proposal make sense?

Reply:
@mvertens I acn do it this way, if you prefer - otherwise, I would have simply forked your repo/branch and would have done a pull request. But as you prefer.

Reply:
@mvertens , ok, I guess, sending the zoom link somehow failed. Above, you were writing that (Quote:) "Multiple compilation errors were found when this was implemented." But ok, I see,  you want to include some more cases for the code included by the pre-processor flags in the `refactor_buildnml`, etc. branches. I will have a look into the code and get back to you.